src/foam/nanos/auth/Group.js

```diff
@@ -45,7 +45,12 @@ foam.CLASS({
       class: 'FObjectArray',
       of: 'foam.nanos.auth.Permission',
       name: 'permissions',
-      documentation: 'Permissions set on group.'
+      documentation: 'Permissions set on group.',
+      preSet: function(oldVal, newVal) {
+        // if old value is undefined, that means save action triggered
+        return oldVal !== undefined ? newVal :
+          newVal.filter(permission => {return Object.keys(permission.instance_).length !== 0});
```

Comment:
As a programmer you shouldn't be using .instance_ directly as that's an internal structure of the objects. Also, it still doesn't ensure that the permission id is set. The id could be empty and the description could be set which would trick this check. Just do 'return permission.id;'